voltdrop_direct/aws-iot-core/index.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * @typedef {Object} DecodedUplink
+ * @property {VoltDropDirectData} data - The open JavaScript object representing the decoded uplink payload when no errors occurred
+ * @property {string[]} errors - A list of error messages while decoding the uplink payload
+ * @property {string[]} warnings - A list of warning messages that do not prevent the driver from decoding the uplink payload
+ */
+
+/**
+ * Decode uplink
+ * @param {Object} input - An object provided by the IoT Flow framework
+ * @param {number[]} input.bytes - Array of bytes represented as numbers as it has been sent from the device
+ * @param {number} input.fPort - The Port Field on which the uplink has been sent
+ * @param {Date} input.recvTime - The uplink message time recorded by the LoRaWAN network server
+ * @returns {DecodedUplink} The decoded object
+ */
+function decodeUplink(input) {
+  const packetList = {
+    VD_DIRECT_VOLTAGE_PF: 40,
+    VD_DIRECT_AMPERAGE: 41,
+    VD_DIRECT_ACT_ENERGY_CONF: 42,
+    VD_DIRECT_ACT_ENERGY_UNCONF: 43,
+    VD_DIRECT_APP_ENERGY_CONF: 44,
+    VD_DIRECT_APP_ENERGY_UNCONF: 45,
+    VD_DIRECT_STARTUP_DIAG: 46,
+    VD_DIRECT_OPERATIONAL_DIAG: 47,
+  };
+
+  // Constant factors for formulas
+  const capacitorVoltageFactor = 5.0 / 255.0;
+  const temperatureCelsiusFactor = 120.0 / 255.0;
+
+  let result = {
+    data: {},
+    errors: [],
+    warnings: [],
+  };
+  const raw = Buffer.from(input.bytes);
+
+  // Uplink payload must be 11 bytes long.
+  if (raw.byteLength != 11) {
+    result.errors.push("Payload length must be 11 bytes");
+    delete result.data;
+    return result;
+  }
+
+  // Packet ID - 1 byte
+  const packetId = raw[0];
+  switch (packetId) {
+    case packetList.VD_DIRECT_VOLTAGE_PF:
+      result.data = {
+        // Average Phase Voltages - 2 bytes each
+        // 16-bit unsigned integers in network byte order (MSB/BE) with 10 integer and 6 fractional bits
+        voltageL1: raw.readUInt16BE(1) / 64.0,
+        voltageL2: raw.readUInt16BE(3) / 64.0,
+        voltageL3: raw.readUInt16BE(5) / 64.0,
+        // Phase Power Factors - 1 byte each, 8-bit signed integers as percentage
+        powerFactorL1: raw.readInt8(7),
+        powerFactorL2: raw.readInt8(8),
+        powerFactorL3: raw.readInt8(9),
+        // Capacitor Voltage Scalar - 1 byte
+        // 8-bit unsigned integer representing the capacitor voltage.
+        // (as if the integer range from 0-255 is scaled to between 0.0V and 5.0V)
+        capacitorVoltage: raw[10] * capacitorVoltageFactor,
+      };
+      if (result.data.capacitorVoltage < 3.4) {
```

Comment:
We can remove this since the VoltDrop core doesn't have a low power mode. This is an oops I included in the original codec that should be removed.